### PR TITLE
added negative numbers, full operations, and edge cases

### PR DESCRIPTION
**Pull request by DiegoDiazGlobant**
Now the calculator works with negative numbers, performs full operations.
Edge cases were also implemented.

Negative numbers and full operations functionality preview:

https://user-images.githubusercontent.com/75440409/105517139-71063400-5cb5-11eb-8990-906f914e3ae4.mp4

Edge cases preview:

https://user-images.githubusercontent.com/75440409/105517039-4d42ee00-5cb5-11eb-86df-06e07d89442e.mp4

### CODE REVIEW

File: app/src/main/java/com/example/calculator/mvp/contract/CalculatorContract.java [hunk 1]
```diff
@@ -1,5 +1,7 @@
 package com.example.calculator.mvp.contract;
 
+import com.example.calculator.utils.ErrorEnum;
+
 public interface CalculatorContract {
```
Comment (SartiMau) [inline app/src/main/java/com/example/calculator/mvp/contract/CalculatorContract.java:5]: Do not abuse with the empty lines between methods pls

File: app/src/main/java/com/example/calculator/mvp/model/CalculatorModel.java [hunk 2]
```diff
@@ -1,12 +1,10 @@
 package com.example.calculator.mvp.model;
 
 import com.example.calculator.mvp.contract.CalculatorContract;
+import com.example.calculator.utils.ErrorEnum;
```
Comment (SartiMau) [inline app/src/main/java/com/example/calculator/mvp/model/CalculatorModel.java]: Remember that we talked about change the name of ErrorUtils for ResultUtils and you can replace the NONE by a SUCCESS

File: app/src/main/java/com/example/calculator/mvp/model/CalculatorModel.java [hunk 3]
```diff
@@ -32,25 +32,35 @@ public void setNewOperand(String value) {
     @Override
     public void setNewOperator(String value) {
         operationValue += value;
-        operator = value;
+        if (firstOperand.isEmpty())
```
Comment (SartiMau) [inline app/src/main/java/com/example/calculator/mvp/model/CalculatorModel.java]: please use curly brackets on each if statement

File: app/src/main/java/com/example/calculator/utils/ErrorEnum.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package com.example.calculator.utils;
+
+public enum ErrorEnum {
+    INCOMPLETE_OPERATION_ERROR,
+    DIVIDE_BY_0_ERROR,
```
Comment (SartiMau) [inline app/src/main/java/com/example/calculator/utils/ErrorEnum.java]: do not use number on the names..change it by DIVIDE_BY_ZERO_ERROR

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/calculator/utils/ErrorEnum.java (hunk 4 above)
Comment (DiegoDiazGlobant) [reply app/src/main/java/com/example/calculator/utils/ErrorEnum.java]: Solved!

File: app/src/main/java/com/example/calculator/mvp/model/CalculatorModel.java (hunk 3 above)
Comment (DiegoDiazGlobant) [reply app/src/main/java/com/example/calculator/mvp/model/CalculatorModel.java]: Solved!

File: app/src/main/java/com/example/calculator/mvp/model/CalculatorModel.java (hunk 2 above)
Comment (DiegoDiazGlobant) [reply app/src/main/java/com/example/calculator/mvp/model/CalculatorModel.java]: Refactored!

File: app/src/main/java/com/example/calculator/mvp/contract/CalculatorContract.java (hunk 1 above)
Comment (DiegoDiazGlobant) [reply app/src/main/java/com/example/calculator/mvp/contract/CalculatorContract.java:5]: solved!

File: app/src/main/java/com/example/calculator/mvp/model/CalculatorModel.java [hunk 5]
```diff
@@ -19,38 +17,52 @@
     private String operationValue = EMPTY;
     public static final int STRING_BEGIN_POSITION = 0;
     public static final int STRING_LAST_POSITION = 1;
+    public boolean equalsPressed = false;
+    private ResultEnum errorEnum;
```
Comment (SartiMau) [inline app/src/main/java/com/example/calculator/mvp/model/CalculatorModel.java]: you forgot to change the variable name

File: app/src/main/java/com/example/calculator/mvp/model/CalculatorModel.java [hunk 6]
```diff
@@ -93,5 +107,35 @@ public void resetValues() {
...
+            operator = EMPTY;
+            secondOperand = EMPTY;
+            operationValue = EMPTY;
+            equalsPressed = false;
+            setNewOperand(result);
+        }
+    }
+
+    @Override
+    public ResultEnum getErrorEnum() {
```
Comment (SartiMau) [inline app/src/main/java/com/example/calculator/mvp/model/CalculatorModel.java]: same here, please change the method's name

**Review by SartiMau — CHANGES_REQUESTED** [review]

**Review by BQuinteros — APPROVED** [review]

File: app/src/main/java/com/example/calculator/mvp/model/CalculatorModel.java (hunk 6 above)
Comment (DiegoDiazGlobant) [reply app/src/main/java/com/example/calculator/mvp/model/CalculatorModel.java]: changed!

File: app/src/main/java/com/example/calculator/mvp/model/CalculatorModel.java (hunk 5 above)
Comment (DiegoDiazGlobant) [reply app/src/main/java/com/example/calculator/mvp/model/CalculatorModel.java]: changed!

**Review by BenjaAbdalaGlobant — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]

**Pull request merged**